README.md

```diff
@@ -0,0 +1,124 @@
+### Modified By: [Saad Hasan](https://github.com/khanyousefzai) and [Hamza Sadruddin](https://github.com/hamza-sadruddin)
+---
+# Introduction
+---
+ Alternate Bit Protocol (ABP) simulator is a communication protocol to ensure reliable transmission through an unreliable network. It consists of three components: sender, network (decomposed further into two subnets) and receiver. The sender sends a packet and waits for an acknowledgement. If the acknowledgement doesn't arrive within a predefined time, the sender re-sends this packet until it receives an expected acknowledgement and then sends the next packet. In order to distinguish two consecutive packets, the sender adds an additional bit on each packet (called alternating bit because the sender uses 0 and 1 alternatively).
+The behavior of receiver is to receive the data and send back an acknowledgement extracted from the received data after a time period. The subnets just pass the packets after a time delay. However, in order to simulate the unreliability of the network, only 95% of the packets will be passed in each of the subnet, i.e. 5% of the data will be lost through the subnet. There is a good documentation for the installation installation of cadmium installation on different platforms like windows and Linux and about Alternate Bit Protocol Simulator. Following the link for more guide.[Installation Guide and Documentation](https://github.com/hamza-sadruddin/AlternateBitProtocolSimulator/tree/master/doc). WE are focuing on the on the proper documentation. reviewing the code, following software documentation and conventions, code standards along with the optimization of the new feature to imporve the simulator.
+
+---
+# Files Organization
+**bin [This folder has files for executables genrated after running the Simulator]**
+- ABP.exe
+- RECEIVER_TEST.exe
+- SENDER_TEST.exe
+- SUBNET_TEST.exe
+
+**build [This Folder has all object generated after after runnung the simulator]**
+- main.o
+- main_receiver.o
+- main_sender.o
+- main_subnet.o
+- message.o
+
+**data  [Data files for testing and some simulator outputs]**
+- receiver_input_test.txt
+- receiver_test_output.txt
+- sender_input_test_ack_In.txt
+- sender_input_test_control_In.txt
+- sender_test_output.txt
+- subnet_input_test.txt
+- subnet_test_output.txt
+
+**doc [This folder contains documents related to ABP and Cadmium including installation guide in winddows and Linux]**
+- alternatebitprot.pdf
+- Cadmium_Documentation_Ubuntu.pdf
+- Cadmium_Windows.pdf
+
+**include [This folder contains the header files and data structures used in the project]**
+1. data_structures [This folder has message.hpp file]
+    - message.hpp
+- receiverCadmium.hpp
+- senderCadmium.hpp
+- subnetCadmium.hpp
+
+**lib [This folder contains 3rd party header files needed in the project]**
+1. cadmium [This folder contains cadmium functionalities]
+2. DESTimes [This folder contains DESTimes functionalities]
+-	iestream.hpp
+-	NDTime.hpp
```

Comment:
Remove the NDTime.hpp file from lib directory, and use the NDTime.hpp file directly from DESTimes/NDTime.hpp as professor mentioned. Accordingly update all paths where NDTiime.hpp is invoked in src and test files.

Reply:
Yes, that's part of Task-D(a). You are mistaken. Please have a look at it again. Thanks.

Reply:
Yes. Sorry, I overlooked it as part-D. 
Part-c comments look good.